src/protocol/commit_strategy.md

```diff
@@ -0,0 +1,21 @@
+# Commit Strategy
+When the `Prover` generates authenticated commitments to the plaintext of the transcript, he can choose which portion(s) of the plaintext to commit to. This dictates the portion(s) that can be selectively disclosed later to the application-specific `Verifier`. This section discusses different commit strategies, as well as their associated costs.
+
+## Strategy
+Given a transcript plaintext of `N` bytes long, the `Prover` can choose from a variety of commit strategies. For example, he can perform one commit to the portion that corresponds to the response body. He can also perform `N` commits to every byte of the transcript. The details of various commit strategies are discussed in the table below.
+
+| # | Strategy | Description | Selective Disclosure | Cost | Usage |
+| - | -------- | ----------- | -------------------- | ---- | ----- |
+| 1 | Commit to custom portion(s) | Only commit to the portion(s) that need to be selectively disclosed later | Cannot selectively disclose anything other than the committed portion(s) | Small artefact size if only a few portions committed | Suitable when only a single (or a few) selective disclosure pattern is needed |
+| 2 | Commit to HTTP objects | Commit to portions that correspond to all common HTTP objects in both request and response, e.g. every header's key and value, every key and value in JSON response body | Flexible to selectively disclose different HTTP objects of transcript at different times | Larger artefact size than strategy #1 | Ideal for most use cases — the default strategy used in the repository's [example](https://github.com/tlsnotary/tlsn/blob/4d5102b6e141ecb84b8a835604be1d285ae6eaa5/crates/examples/attestation/prove.rs#L99) |
```

Comment:
This points to a specific commit. Do we dare to link to the latest (dev) version instead?

Reply:
It's pointing to a specific line, so i used a permalink instead since the line is very likely to be moved as soon as we make changes to the example